src/borg/archiver.py

```diff
@@ -1222,6 +1226,116 @@ def do_debug_dump_archive_items(self, args, repository, manifest, key):
         print('Done.')
         return EXIT_SUCCESS
 
+    def debug_json(self, d):
+        def decode_bytes(value):
+            # this should somehow be reversable later, but usual strings should
+            # look nice and chunk ids should mostly show in hex. Use a special
+            # inband signaling character (ASCII DEL) to distinguish between
+            # decoded and hex mode.
+            if not value.startswith(b'\x7f'):
+                try:
+                    value = value.decode()
+                    return value
+                except UnicodeDecodeError:
+                    pass
+            return '\u007f' + bin_to_hex(value)
+
+        def decode_tuple(t):
+            res = []
+            for value in t:
+                if isinstance(value, dict):
+                    value = decode(value)
+                elif isinstance(value, tuple) or isinstance(value, list):
+                    value = decode_tuple(value)
+                elif isinstance(value, bytes):
+                    value = decode_bytes(value)
+                res.append(value)
+            return res
+
+        def decode(d):
+            res = collections.OrderedDict()
+            for key, value in d.items():
+                if isinstance(value, dict):
+                    value = decode(value)
+                elif isinstance(value, tuple) or isinstance(value, list):
+                    value = decode_tuple(value)
+                elif isinstance(value, bytes):
+                    value = decode_bytes(value)
+                if isinstance(key, bytes):
+                    key = key.decode()
+                res[key] = value
+            return res
+
+        return decode(d)
```

Comment:
I didn‘t want to bloat the helpers module even more with quite specific functions. decode_bytes is not really general purpose and is referenced by this code.